core/src/main/java/org/geysermc/geyser/translator/text/MessageTranslator.java

```diff
@@ -428,15 +433,95 @@ public static String normalizeSpace(String string) {
     /**
      * Deserialize an NbtMap provided from a registry into a string.
      */
-    // This may be a Component in the future.
-    public static String deserializeDescription(NbtMap tag) {
+    public static String deserializeDescription(GeyserSession session, NbtMap tag) {
         NbtMap description = tag.getCompound("description");
-        String translate = description.getString("translate", null);
-        if (translate == null) {
-            GeyserImpl.getInstance().getLogger().debug("Don't know how to read description! " + tag);
-            return "";
+        Component parsed = componentFromNbtTag(description);
+        return convertMessage(session, parsed);
+    }
+
+    public static Component componentFromNbtTag(Object nbtTag) {
+        return componentFromNbtTag(nbtTag, Style.empty());
+    }
+
+    private static Component componentFromNbtTag(Object nbtTag, Style style) {
+        if (nbtTag instanceof String literal) {
+            return Component.text(literal).style(style);
+        } else if (nbtTag instanceof List<?> list) {
+            return Component.join(JoinConfiguration.builder(), componentsFromNbtList(list, style));
```

Comment:
I think `JoinConfiguration.noSeparators()` is the equivalent to `builder()` without needing to build an empty JoinConfiguration, judging from the code.